src/dataflow-types/src/types.rs

```diff
@@ -289,1159 +281,1229 @@ impl<View> DataflowDescription<View> {
+            pub descriptors: Vec<u8>,
+            pub message_name: NormalizedProtobufMessageName,
+            pub schema_registry_config: Option<ccsr::ClientConfig>,
+        }
 
-/// Arguments necessary to define how to decode from CSV format
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct CsvEncoding {
-    pub columns: ColumnSpec,
-    pub delimiter: u8,
-}
+        /// Arguments necessary to define how to decode from CSV format
+        #[derive(Clone, Debug, Serialize, Deserialize)]
+        pub struct CsvEncoding {
+            pub columns: ColumnSpec,
+            pub delimiter: u8,
+        }
 
-/// Determines the RelationDesc and decoding of CSV objects
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub enum ColumnSpec {
-    /// The first row is not a header row, and all columns get default names like `columnN`.
-    Count(usize),
-    /// The first row is a header row and therefore does become data
-    ///
-    /// Each of the values in `names` becomes the default name of a column in the dataflow.
-    Header { names: Vec<String> },
-}
+        /// Determines the RelationDesc and decoding of CSV objects
+        #[derive(Clone, Debug, Serialize, Deserialize)]
+        pub enum ColumnSpec {
+            /// The first row is not a header row, and all columns get default names like `columnN`.
+            Count(usize),
+            /// The first row is a header row and therefore does become data
+            ///
+            /// Each of the values in `names` becomes the default name of a column in the dataflow.
+            Header { names: Vec<String> },
+        }
 
-impl ColumnSpec {
-    /// The number of columns described by the column spec.
-    pub fn arity(&self) -> usize {
-        match self {
-            ColumnSpec::Count(n) => *n,
-            ColumnSpec::Header { names } => names.len(),
+        impl ColumnSpec {
+            /// The number of columns described by the column spec.
+            pub fn arity(&self) -> usize {
+                match self {
+                    ColumnSpec::Count(n) => *n,
+                    ColumnSpec::Header { names } => names.len(),
+                }
+            }
+
+            pub fn into_header_names(self) -> Option<Vec<String>> {
+                match self {
+                    ColumnSpec::Count(_) => None,
+                    ColumnSpec::Header { names } => Some(names),
+                }
+            }
         }
-    }
 
-    pub fn into_header_names(self) -> Option<Vec<String>> {
-        match self {
-            ColumnSpec::Count(_) => None,
-            ColumnSpec::Header { names } => Some(names),
+        #[derive(Clone, Debug, Serialize, Deserialize)]
+        pub struct RegexEncoding {
+            #[serde(with = "serde_regex")]
+            pub regex: Regex,
         }
     }
-}
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct RegexEncoding {
-    #[serde(with = "serde_regex")]
-    pub regex: Regex,
-}
+    pub mod persistence {
```

Comment:
I would likely not put `BringYourOwn`, `Consistency`, and `TimestampSourceUpdate` in `mod persistence` but I think I can see the motivation.

Reply:
Yes, I don't know what they are. I believe at least `TimestampSourceUpdate` exists in order to record timestamp bindings, in the name of persistence (nb not `persist` crate).